src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -264,11 +264,12 @@ public void WriteConnectionString(IResource resource)
     /// collection.
     /// </summary>
     /// <param name="resource">The <see cref="IResource"/> that contains <see cref="EndpointAnnotation"/> annotations.</param>
-    /// <param name="emitContainerPort">Flag to determine whether container port is emitted.</param>
-    public void WriteBindings(IResource resource, bool emitContainerPort = false)
+    public void WriteBindings(IResource resource)
     {
         if (resource.TryGetEndpoints(out var endpoints))
         {
+            var allocator = new PortAllocator();
```

Comment:
I'm wondering whether this should be more global in scope. If the target compute platform is something like Docker compose you could end up with port conflicts for the external ports right?

Reply:
This is a *GREAT* question.  I think this does have to be global. We probably shouldn't assume anything.

Reply:
Done